### PR TITLE
feat(agent): add options for create agent CDAAS-2127

### PR DESCRIPTION
**Pull request by th3morg**
Mostly for use with integration tests, options have been added to the `create agent` command

## How has this been tested? How can a reviewer test these changes?

# Have your performed the following tests?

Please confirm you have done the following tests to ensure your pull request is ready to be merged..

- [ ] If this change modifies the deployment flow have triggered a deployment using this branch (i.e. `build/dist/darwin_amd64/armory deploy start -f <path to file>`)?
- [ ] If this change modifes the login flow have you tried logging in and out with this branch (i.e. `build/dist/darwin_amd64/armory login`) and verified the credentials work?
- [ ] Have you verified the specific change made in this PR?

### CODE REVIEW

File: cmd/agent/agent_create.go [hunk 1]
```diff
@@ -75,25 +87,30 @@ func NewCmdCreateAgent(configuration *config.Configuration) *cobra.Command {
...
-			if err := o.Run(); err != nil {
+			if err := options.Run(); err != nil {
 				return err
 			}
 			return nil
 		},
 		SilenceUsage: true,
 	}
+
+	cmd.Flags().StringVarP(&options.ContextName, "context-name", "", "", "specify the name of the kubernetes context to select from your kube config")
```
Comment (danielpeach) [inline cmd/agent/agent_create.go]:
```suggestion
	cmd.Flags().StringVarP(&options.ContextName, "context-name", "", "", "specify the name of the Kubernetes context to select from your kubeconfig")
```

The Kubernetes documentation calls it a "kubeconfig" (one word): https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/

Comment (danielpeach) [reply cmd/agent/agent_create.go]: This one doesn't say "Skips prompt" but I would have thought that it does?

**Review by danielpeach — APPROVED** [review]
I pulled this down and verified that it doesn't break the old flow (also tried out the flag-based flow).

For anyone looking in on this PR - the ticket that Ryan is working on is to add testing to `armory agent create`. This change is in service of adding integration tests to the feature. Please forgive him for not including unit tests.

File: cmd/agent/agent_create.go (hunk 1 above)
Comment (th3morg) [reply cmd/agent/agent_create.go]: Yes, yes it does :-D

**Pull request merged**